### ISSUE TITLE
Index invalid or out of range

### ISSUE THREAD

**Issue by JanGalek** [nette/coding-standard#12]
When File is empty, than ecs return error:
```[RuntimeException]
Index invalid or out of range```

maybe checking filesize / ignore / remove ?

**Comment by TomasVotruba** [comment]
Intereseting experiment :) 

Like empty PHP file without header `<?php`?

**Comment by JanGalek** [comment]
yes, I only prepared file to do and run check :)

**Comment by TomasVotruba** [comment]
@JanGalek I'm on it!

**Comment by TomasVotruba** [comment]
Resolved by https://github.com/Symplify/Symplify/commit/4e3384bdade816aa78d5455c7a92afc75345bd57

**Comment by TomasVotruba** [comment]
Resolved since Symplify\EasyCodingStandard 2.2.2

**Comment by JanGalek** [comment]
Nice, thank you :)

**Comment by dg** [comment]
This code https://github.com/Symplify/Symplify/commit/4e3384bdade816aa78d5455c7a92afc75345bd57 is imho not used by nette coding standard.

**Comment by TomasVotruba** [comment]
You're right. Could you add it to CustomSourceProvider?

**Comment by dg** [comment]
Is not it a bad place to fix? This should not be the responsibility of the SourceFinder.

**Comment by TomasVotruba** [comment]
I was thinking about it. As more places can depend on source provider, I think it is. Empty files should be processed and provided anywhere.

Where would you fix it?

**Comment by dg** [comment]
Why could not the file be empty?

**Comment by TomasVotruba** [comment]
Because there is no content coding standard can be applied to.

**Comment by dg** [comment]
But it should not trigger error.

**Comment by TomasVotruba** [comment]
Mm, after bit investigation it looks like this is the exception in the Finder part, that has nothing to do with empty files.

I will investigave in couple of days.

**Comment by TomasVotruba** [comment]
So I found out [php-cs-fixer drops empty files as well](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3d74d5fc62beebb93ed334c7a9515cbd4d9d4773/src/Runner/FileFilterIterator.php#L73).

Without that, most fixers would fail on array access errors even in `isCandidate()` method, that is shown there, e.g. [BracesFixer here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3d74d5fc62beebb93ed334c7a9515cbd4d9d4773/src/Fixer/Strict/DeclareStrictTypesFixer.php#L61)

**Comment by dg** [comment]
It should be filtered on higher level, there is no reason that CustomSourceProvider must not provide files that are prefectly valid but some filters are "broken". What about `array_filter('filesize', $files)` in SourceFinder?

**Comment by dg** [comment]
Size() must be before from()

**Comment by TomasVotruba** [comment]
@dg I see. 
In that case I'm unable to influence already created Finder with from(), so I used the `array_filter($files, 'filesize')` filter you suggested. I don't need that now, but it should inform me by error or independent.

**Comment by TomasVotruba** [comment]
Ready to review: https://github.com/Symplify/Symplify/pull/287

**Comment by TomasVotruba** [comment]
Is fixed now :+1:

**Comment by TomasVotruba** [comment]
Released in 0.5: https://github.com/nette/coding-standard/releases/tag/v0.5.0
